flowey/flowey_core/src/pipeline.rs

```diff
@@ -947,6 +949,13 @@ impl PipelineJob<'_> {
         self
     }
 
+    /// Overrides the name of the job. The main use case for this is to get an
+    /// artifact in ADO with a specific name.
```

Comment:
Rather than prescribing a use-case, I would just document precisely what this method does, and let callers add extra flavor around why they are doing the override.

so, in this case, I would just have this method's docs talk about how typically, flowey will generate a reasonable job ID without the user having to care about it, but in rare occasions (e.g: interfacing with existing templates that rely on the current job ID), it may be useful to override it to something custom.

Reply:
I would try to convey the message that specific job IDs are not typically user-visible, and that unless you really know what you're doing, its best to just have a descriptive display_name when instantiating the job.